plugins/outputs/otlp/client.go

```diff
@@ -0,0 +1,314 @@
+package otlp
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	metricsService "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// serviceConfig copied from OTel-Go.
+	// https://github.com/open-telemetry/opentelemetry-go/blob/5ed96e92446d2d58d131e0672da613a84c16af7a/exporters/otlp/grpcoptions.go#L37
+	serviceConfig = `{
+	"methodConfig":[{
+		"name":[
+			{ "service":"opentelemetry.proto.collector.metrics.v1.MetricsService" },
+			{ "service":"opentelemetry.proto.collector.trace.v1.TraceService" }
+		],
+		"retryPolicy":{
+			"MaxAttempts":5,
+			"InitialBackoff":"0.3s",
+			"MaxBackoff":"5s",
+			"BackoffMultiplier":2,
+			"RetryableStatusCodes":[
+				"UNAVAILABLE",
+				"CANCELLED",
+				"DEADLINE_EXCEEDED",
+				"RESOURCE_EXHAUSTED",
+				"ABORTED",
+				"OUT_OF_RANGE",
+				"UNAVAILABLE",
+				"DATA_LOSS"
+			]
+		}
+	}]
+}`
```

Comment:
As far as I can see this is a one-to-one copy, so why don't you use the variable exported by the cited source:
```suggestion
	serviceConfig = optl.DefaultGRPCServiceConfig
```
Please note: You need to import `import "go.opentelemetry.io/otel/exporters/otlp"` to have access to this variable.

Reply:
I've updated the link, the configuration was moved to an internal package: https://github.com/open-telemetry/opentelemetry-go/blob/a2cecb6e80f6a0712187b080a97f8efb5a61082a/exporters/otlp/internal/otlpconfig/options.go#L47

Reply:
Forgive me if this is a dumb comment, but I'm not deep into open-telemetry... Reading the new referenced version I think you can use [`NewDriver()`](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp@v0.19.0/otlpgrpc#NewDriver) and then call `ExportMetrics()` and `ExportTraces()` on [the resulting driver](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp@v0.19.0#ProtocolDriver)... Am I missing something?

Reply:
The metrics API and SDK are still being defined so i'd rather not rely on them just yet for this code. Once it's solidified, i'll likely refactor this plugin to use those instead.

Reply:
Hmm, it's versioned as 0.19, so I think you can rely on this. I think the interface of 0.19 referenced to above
a) simplifies the code massively
b) does not need copying internal code from the client that might be subject to change
c) should be guaranteed to work by the maintainers of the package

So to be honest I would feel better to rely on their code and maybe later change if 0.20, 1.0 or whatever comes out...